counting_sort/counting_sort.go

```diff
@@ -0,0 +1,35 @@
+package main
+
+import "fmt"
+
+// CountingSort sorts the input array in ascending order
+// Time complexity is O(z) where z =  max(len(data), max(data))
+func CountingSort(data []int) {
+	temp := make([]int, len(data))
```

Comment:
You can declare a variable as `len(data)`, as you have used it 3 times.
and then use that variable in place of `len(data)`